src/ui/component/toast/toast.tsx

```diff
@@ -1,52 +1,121 @@
 import classNames from 'classnames'
-import { useEffect } from 'react'
+import type { FC } from 'react'
+import { useCallback, useRef, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import * as ToastPrimitive from '@radix-ui/react-toast'
+import type { ActionType, NotificationID, NotificationType } from '@/domain/notification/entity'
+import type { IconName } from '@/ui/component/icon/icon'
+import { Icon } from '@/ui/component/icon/icon'
+import { useOnClickOutside } from '@/ui/hook/useOnClickOutside'
+import { useOnKeyboard } from '@/ui/hook/useOnKeyboard'
+import { Button } from '@/ui/component//button/button'
+import type { DismissNotificationInput } from '@/domain/notification/aggregate'
+import '@/ui/component/notification/i18n'
 import './toast.scss'
 
-type ToastVariants = 'success' | 'error' | 'warning' | 'info'
+const actionsEffects: Record<ActionType, () => void> = {
+  refresh: () => window.location.reload()
+}
+
+type ToastCTAProps = {
+  type: NotificationType
+  action: ActionType
+}
+const ToastCTA: FC<ToastCTAProps> = ({ action, type }) => {
+  const { t } = useTranslation('notification')
+
+  const handleClick = useCallback(() => actionsEffects[action](), [action])
+  return (
+    <Button
+      className={classNames('okp4-dataverse-portal-toast-cta', type)}
+      label={t(`action.${action}`)}
+      onClick={handleClick}
+    />
+  )
+}
 
 type ToastProps = {
+  action?: ActionType
+  id: NotificationID
+  type: NotificationType
   title: string
-  autoHideDuration?: number
-  description?: string
-  icon?: JSX.Element
-  onClose?: () => void
-  open?: boolean
-  variant?: ToastVariants
+  onDismiss: (input: DismissNotificationInput) => void
+  message?: string
+  iconName?: IconName
+  duration: number
 }
 
-export const Toast = ({
+// eslint-disable-next-line max-lines-per-function
+export const Toast: FC<ToastProps> = ({
+  action,
+  id,
+  type,
   title,
-  autoHideDuration,
-  description,
-  icon,
-  onClose,
-  open = false,
-  variant = 'info'
-}: ToastProps): JSX.Element | null => {
-  useEffect(() => {
-    if (open && autoHideDuration) {
-      const autoHideDurationTimer = setTimeout(() => {
-        onClose?.()
-      }, autoHideDuration)
-      return () => {
-        clearTimeout(autoHideDurationTimer)
-      }
+  onDismiss,
+  message,
+  iconName,
+  duration
+}) => {
+  const [isOpen, setIsOpen] = useState(true)
+  const toastRootRef = useRef<HTMLLIElement | null>(null)
+
+  const onClose = useCallback(() => {
+    if (isOpen) {
+      const timeout = setTimeout(() => {
+        clearTimeout(timeout)
```

Comment:
`clearTimeout(timeout)` should be outside setTimeout, no?

Reply:
This is not a conventional but valid approach!
The `clearTimeout(timeout)` is placed before the `onDismiss({ id })` call, which ensures that the timeout is cleared when the parent component causes this component to be unmounted.